### DISCUSSION TITLE
Multiple Docker Services Config Question

### DISCUSSION THREAD

**Discussion by dm17** [nginx-proxy/nginx-proxy#2105] [Q&A]
I realize that the correct /etc/nginx/conf.d/default.conf should be automatically generated. However, I'm confused about how the VIRTUAL_HOST & VIRTUAL_PORT vars from my 3 containers (that also have port: "host:container" defined) get mapped to nginx's 80 & 443. Inside of docker my 3 containers talk to each other on 3 ports - and they all need to talk to the internet as well. Each service talks to the other services using its docker service name - and each service name is also in /etc/hosts. This all had it all working well in dev - but not production (the difference being nginx + letsencrypt). Perhaps each service needs a different domain name? 
Any debugging suggestions?

I was recommended in this thread to ask here:
https://github.com/JrCs/docker-letsencrypt-nginx-proxy-companion/issues/565

Thanks.

**Comment by cron410** [comment]
This container has read access to your Docker socket, so it can retrieve information about other containers from it. You specify what hostname e.g. `server1.yourdomain.com` in each container's run command, or in each container's Compose directives. These labels are picked up by the proxy container, along with the port they are listening on and the NGINX proxy config is built automatically.

Let's say you have a container hosting your blog, on port 7080. You want users hitting your host with `blog.yourdomain.com` on port 80 to be served by that container. You also have a container serving RSS or something similar on port 9080, and users would reach that with `rss.yourdomain.com`.  Here's some example configs:

docker run -d \
 --name ghostblog \
 -e VIRTUAL_HOST=blog.yourdomain.com \
 -e url=http://blog.yourdomain.com:7080 \
-p 7080:7080 \
ghost

docker run -d \
    --name=rss2email \
   -e VIRTUAL_HOST=rss.yourdomain.com
    --volume=/path/to/data:/var/lib/rss2email \
    blalor/rss2email

docker run -d \
  -p 80:80 \
  -v /var/run/docker.sock:/tmp/docker.sock:ro \
  jwilder/nginx-proxy
